Applications/Spire/Source/Scratch/main.cpp

```diff
@@ -1,12 +1,105 @@
 #include <QApplication>
```

Comment:
Revert changes to this and incorporate this demo to the best that you can into the Ui Viewer. The UI Viewer doesn't need to have an exact copy of this demo but it should be good enough to test out most functionality.